src/bin/stratis_dumpmetadata.rs

```diff
@@ -12,15 +12,19 @@ use std::{
 
 use serde_json::Value;
 
-use libstratis::engine::BDA;
+use libstratis::engine::{StaticHeader, BDA};
 
 fn run(devpath: &str) -> Result<(), String> {
     let mut devfile = OpenOptions::new()
         .read(true)
         .open(&devpath)
         .map_err(|the_io_error| format!("Error opening device: {}", the_io_error))?;
 
-    let bda = BDA::load(&mut devfile)
+    let read_results = StaticHeader::read_sigblocks(&mut devfile);
+    println!("{:#?}", read_results);
+    let header =
```

Comment:
You'll want to scrutinize the error messages here a bit. If there was an error just reading the bytes, then you don't know that a BDA was detected.

This might be a reason to change the signature of BDA::load(). Right now, one parameter is ```StratisResult<Option<StaticHeader>>```. Maybe it should just be ```Option<StaticHeader>``` or ```StaticHeader```. Try having the argument be just ```StaticHeader```, and see how that goes.

Reply:
Done!